src/Analysis/Ast/Test/ImportTests.cs

```diff
@@ -234,5 +237,19 @@ import sys
             var analysis = await GetAnalysisAsync(code);
             analysis.Should().HaveVariable("x").OfType(BuiltinTypeId.Int);
         }
+
+        [TestMethod, Priority(0)]
+        public async Task ModuleMembers() {
+            var appUri = TestData.GetTestSpecificUri("app.py");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("package", "__init__.py"), "import m1");
+            await TestData.CreateTestSpecificFileAsync(Path.Combine("package", "m1", "__init__.py"), string.Empty);
+
+            await CreateServicesAsync(PythonVersions.LatestAvailable3X);
+            var rdt = Services.GetService<IRunningDocumentTable>();
+            var doc = rdt.OpenDocument(appUri, "import package");
+
+            var analysis = await doc.GetAnalysisAsync(Timeout.Infinite);
+            analysis.Should().HaveVariable("package").Which.Should().HaveMember("m1");
```

Comment:
This is wrong. 
This example works (output: `module 'package.m1' from 'D:\\Temp\\package\\m1\\__init__.py'`):
```
from package import m1
print(m1)
```

This example doesn't (output: `AttributeError: module 'package' has no attribute 'm1'`):
```
import package
print(package.m1)
```

Reply:
There are tests `InitPyVsModuleNameImport_AbsoluteImport` and `PackageModuleImport` that show correct behavior

Reply:
Yeah, I think we're going to have to handle this another way, like ensuring that modules make it out as members of the module. Since that's going to get big (with every module that some other module imports as a member), maybe we can do something like "export it if it's a module below this one", which would probably handle many common cases of libraries re-exporting submodules.

Reply:
Submodules and subpackages are available as members only if they either are exported from module or package's `__init__.py` or if they are imported by previous code (that is what `PythonVariableModule` is for):
```py
from package import m1
import package
print(package.m1)
```

Reply:
@jakebailey , I don't think there is anything we can address. Original bug #1301 has issue with `import torch` probably because in `torch/__init__.py` it doesn't list `nn` in the `__all__` explicitly.

Reply:
`__all__` is only applicable for `*` imports. It does not affect member accesses, only what is copied into the scope when doing something like `from torch import *`.

Reply:
I think the actual issue is that we don't offer modules in other modules as members, but getting that right while not also being annoying is tricky.

Reply:
True but if user types it anyway we don't squiggle member saying it does not exist. Specific case is in the bug, i.e. people want `pandas.testing` to show up so they can type `pandas.testing.assert`

Reply:
But it may not exist, so showing it is going to be misleading since it'd crash at runtime.

Reply:
Yep, I was wrong about `__all__`. The problem is that we ignore modules that are imported by `torch` itself or any other modules. If I change `package/__init__.py` to 
```py
import package.m1
```
then
```py
import package
print(package.m1)
```
will work despite the fact that there is no member `m1` in `package`.

Reply:
This information is collected in `ImportHandler._variableModules` during analysis, but it isn't shared between modules. And passing it through is tricky indeed. For example, if we add a module `package/m2.py`:
```py
from package import m1 as p
```
and use the following code in `package/__init__.py`:
```py
import package.m2
```
then 
```py
import package
print(package.m1)
```
will work again.

Reply:
After some experimentation, the example of `import package.m1` inside of `package`'s `__init__.py` causing `m1` to be available to importers of `package` via `import package` as `package.m1` is definitely true; we're seeing it for some parts of `pandas` as well. Seems like that needs to be fixed as well as making sure that they're saved as members for completion.

Reply:
![image](https://user-images.githubusercontent.com/5341706/65274567-74dde100-dad8-11e9-9b6e-e1108c07ff65.png)

Where `top/__init__.py` is:

```python
from top import sub1

import top.sub2
import top.sub3.sub4
```

Running:

```python
import top
print(dir(top))
```

Gives:
```
[
    '__builtins__',
    '__cached__',
    '__doc__',
    '__file__',
    '__loader__',
    '__name__',
    '__package__',
    '__path__',
    '__spec__',
    'sub1',
    'sub2',
    'sub3',
    'top'
]
```

Reply:
So essentially, if you make an import within your module of a submodule, then there's a member that's added to that module for the submodule, such that you can access it externally, but no more than that single level.

Reply:
Done